### ISSUE TITLE
Forced URL scheme for multilanguage and multiversion 

### ISSUE THREAD

**Issue by mart-e** [jdillard/sphinx-sitemap#22]
Hello,

Reading the code, I understand that you assume the page is published, using the version and language configuration, following the scheme : 
- `https://<html_baseurl>/<language>/<version>/<page>.html`

https://github.com/jdillard/sphinx-sitemap/blob/02bbe94613ca1164a0f73865659d240fadb3f808/sphinx_sitemap/__init__.py#L103-L104

Is there a reason to force this construction? For instance, on [our website](https://www.odoo.com/documentation/user/), we use the construction: 
- `https://<base>/<version>/<language>/<page>/html`

What would be the way to make it work for our scheme?

Thanks

**Comment by humitos** [comment]
This assumption probably comes from Read the Docs, since this project was created after a discussion there https://github.com/readthedocs/readthedocs.org/issues/557#issuecomment-377420985

**Comment by mart-e** [comment]
Thanks for the reply.

@jdillard Would you accept a patch for a configurable url? For instance something like:

```python
scheme = app.config.i18n_url_scheme or "/{lang}/{version}/{link}"
url = site_url + scheme.format(lang=app.builder.config.language, version=version, link=link)
```

**Comment by jdillard** [comment]
Correct, I had made it opinionated based on RTD's scheme just to get something out. Although, it looks like I forgot to make an issue to remind myself to circle back and fix it, nice catch and nice fix!

PR's gladly accepted! I just did a quick test and that looks like that solution will work, it just may need some finagling to get everything aligned again. If you want to submit a first draft I can help clean it up if it needs any.

**Comment by jdillard** [comment]
I merged your changes in and then realized it forced a multilanguage and multiversion scheme as a default and didn't like that so changed the behavior to only include `language` and `version` if they exist and auto appended a trailing slash to each. You can still re-arrange them, you just don't need slashes anymore. I also changed the new config value name to `sitemap_url_scheme` since language is optional now.

That is all documented in the README now if you would like to test the master branch before I cut a new version.

**Comment by mart-e** [comment]
Great, thanks for the addition and for the adaptation !

**Comment by jdillard** [comment]
Thanks for the contribution, this was a great step forward! I went ahead and released it as version `2.0.0` since it has the possibility to be a breaking change for lower versions.

I already tested it with some unique schemes I have in other projects and the configuration is much cleaner now.
